libdrgn/error.c

```diff
@@ -117,6 +119,36 @@ LIBDRGN_PUBLIC struct drgn_error *drgn_error_format(enum drgn_error_code code,
 	return drgn_error_create_nodup(code, message);
 }
 
+LIBDRGN_PUBLIC struct drgn_error *drgn_error_format_read(enum drgn_error_code code,
+						         uint64_t address,
+						         const char *format, ...)
+{
+	struct drgn_error *err;
+	char *message;
+	va_list ap;
+	int ret;
+
+	va_start(ap, format);
+	ret = vasprintf(&message, format, ap);
+	va_end(ap);
+
+	if (ret == -1)
+                return &drgn_enomem;
+
+	err = drgn_error_create_nodup(code, message);
+	if (err == &drgn_enomem)
+		return err;
+
+	err->address = malloc(sizeof(*err->address));
```

Comment:
Why does this need to be malloc'd? Can't we just make `err->address` a `uint64_t` rather than a `uint64_t *`?

If the concern is distinguishing between an address of 0 and an unspecified address, then let's add the "out of bounds" error first so that the address is always specified here.